interfaces/builtin/fwupd.go

```diff
@@ -64,13 +68,78 @@ const fwupdPermanentSlotAppArmor = `
   /sys/firmware/efi/esrt/entries/** r,
 
   # Allow fwupd to access system information
-  /sys/devices/virtual/dmi/id/product_name r,
-  /sys/devices/virtual/dmi/id/sys_vendor r,
+  /sys/firmware/dmi/tables/smbios_entry_point r,
+  /sys/devices/virtual/dmi/id/{sys,bios,product,chassis,board}_{name,version,vendor,date,serial,sku,type,asset_tag,family} r,
+
+  /sys/firmware/dmi/tables/DMI r,
+  /sys/bus/usb/devices/ r,
+
+  /sys/firmware/acpi/bgrt/{type,version,{x,y}offset,image} r,
 
   # Allow read/write access for efivarfs filesystem
   /sys/firmware/efi/efivars/ r,
   /sys/firmware/efi/efivars/** rw,
 
+  /proc/modules r,
+  /run/udev/data/* r,
+  /sys/devices/**/usb*/busnum r,
+  /sys/devices/**/usb*/**/busnum r,
+  /sys/devices/**/usb*/devnum r,
+  /sys/devices/**/usb*/**/devnum r,
+  /sys/devices/**/usb*/speed r,
+  /sys/devices/**/usb*/**/speed r,
+  /sys/devices/**/usb*/descriptors r,
+  /sys/devices/**/usb*/**/descriptors r,
+  /sys/devices/**/usb*/bConfigurationValue r,
+  /sys/devices/**/usb*/**/bConfigurationValue r,
+  /dev/mem r,
+  /sys/kernel/security/lockdown r,
+  owner @{PROC}/@{pid}/mountinfo r,
+  /sys/bus/pci/devices/ r,
+  /sys/devices/pci*/**/vendor r,
+  /sys/devices/pci*/**/device r,
+  /sys/devices/pci*/**/revision r,
+  /sys/devices/pci*/**/subsystem_vendor r,
+  /sys/devices/pci*/**/subsystem_device r,
+  /sys/devices/pci*/**/class r,
+  /sys/devices/pci*/**/config r,
+  /sys/devices/pci*/**/name r,
+  /sys/devices/**/serio*/description r,
+  /dev/cpu/*/msr rw,
+
+  /sys/devices/**/description r,
+
+  /sys/devices/**/mei/*/fw_ver r,
+  /sys/devices/**/unique_id r,
+
+  /sys/devices/**/nvm_version r,
+  /sys/devices/**/generation r,
+  /sys/devices/**/vendor_name r,
+  /sys/devices/**/nvmem r,
+  /sys/devices/**/authorized r,
+
+  capability sys_rawio,
+  capability sys_nice,
+
+  owner @{PROC}/@{pid}/mounts r,
+  /run/mount/utab r,
+  /sys/bus/*/devices/ r,
+  /sys/devices/**/i2c*/**/name r,
+
+  /sys/devices/**/tpm/tpm*{,/**} r,
+  /sys/kernel/security/tpm*/binary_bios_measurements r,
+  /dev/tpm* rw,
+  /dev/tpmrm* rw,
+  /sys/firmware/acpi/tables/FACP r,
+  /sys/power/mem_sleep r,
+  /proc/swaps r,
+  /proc/sys/kernel/tainted r,
+
+  /dev/mei[0-9]* rw,
+  /dev/nvme[0-9]* rw,
+  /dev/gpiochip[0-9]* rw,
+  /dev/drm_dp_aux[0-9]* rw,
+
```

Comment:
You're missing `/dev/block` which is used by the Android bootloaders plugin.

Reply:
Would not be a symlink? We need the device node here.

Reply:
@DylanVanAssche can you add more detail here on the paths the symlinks point to from a system supporting `android-boot`?

Reply:
`/dev/block` can point to any block device such as:

- `/dev/sdX` where `X` is a character such as `/dev/sda`, `/dev/sdb`, etc.
- `/dev/mmcblkX` where `X` is a number such as `/dev/mmcblk0`, `/dev/mmcblk1`.
- `/dev/mmcblkXbootY` where `X` and `Y` are numbers such as `/dev/mmcblk0boot0`
- ...

All of these are created by the kernel for block devices.

Reply:
I suppose we can add `block-devices` plug then. And add `/dev/mmcblkXbootY` to the `block-devices` interfaces.

Thank you.

Reply:
Do we need partition block devices? Or whole disks only?

Reply:
I have added `/dev/mmcblk[0-9]{,[0-9],[0-9][0-9]}boot[0-9]* rwk` and `/sys/devices/**/mmcblk[0-9]{,[0-9],[0-9][0-9]}boot[0-9]*/force_ro rw` to fwupd interface. I supposed that unlocking read only is needed.

Reply:
@valentindavid I notice you suggested `block-devices` above - I tend to agree that using this would make the intent clearer and perhaps more future-proof for fwupd - in that case are these `/dev/mmcblk` entries still needed here?

Reply:
The problem is that `block-devices` does not give access to partitions, but to full disks. That means `mmcblk*boot*` are not accessible with `block-devices`. So we still need it.

Reply:
`mmcblk*boot*` and specially `force_ro` might be dangerous to give to all snaps with `block-devices`. My understanding is that MMC has special boot partitions to protect from writing on it by mistake. So in that spirit, we should not give access to applications that do not need to flash it.